### ISSUE TITLE
backend app.py not functioning properly

### ISSUE THREAD

**Issue by khmorad** [khmorad/recepieMakerAI#28]
when running app.py  vgg16_final.keras gets downloaded however there is still problem with the application trying to find the .keras file

## when ingredient image is uploaded:
![image](https://github.com/user-attachments/assets/9065a464-8f98-4ed2-83d9-9630077cfdcf)

**Comment by tkpp26** [comment]
ok, thanks for letting me know! it worked on my local, but let me check it out!

**Comment by tkpp26** [comment]
Do you see a file called vgg16_final.keras in ur back directory after download?

**Comment by khmorad** [comment]
## it seems like the vgg16_final.keras file gets downloaded to the root directory of my project I'm still facing the same issue
![image](https://github.com/user-attachments/assets/07a3801c-2234-4dbe-a2fe-8679b4635d3f)

## location where the file is downloaded:
![image](https://github.com/user-attachments/assets/29056582-5ca2-442b-ba61-916adc0f617e)

**Comment by tkpp26** [comment]
hmm, yea i think it should be downloaded in ur back directory...I wonder why.
u did python3 app.py from ur back directory, am i correct?

[
<img width="195" alt="Screenshot 2024-09-17 at 15 45 35" src="https://github.com/user-attachments/assets/e4ebe7f1-752e-4248-8fa7-1618f8129d40">
](url)

**Comment by khmorad** [comment]
i use `python app.py` command `python3 app.py` is not recognized in my computer does that matter?......hmm maybe something is wrong on my side I can clone once again I will let you know

**Comment by khmorad** [comment]
so i cloned the project to the new directory and the uploads folder is generated in the root directory and when I upload image the vgg file is not being downloaded anymore if there is a way I could fix it let me know!

![image](https://github.com/user-attachments/assets/06c4d9ea-9541-4bdf-b0b6-72ace8cfd4e0)

## after uploading image
![image](https://github.com/user-attachments/assets/6811d22e-7a37-4433-9580-b5afb80588d3)

**Comment by tkpp26** [comment]
i think python app.py should be fine! hmm.. the upload directory should also be created in the back directory. Just to make sure again, u're running app.py when u are in ur back directory?

**Comment by tkpp26** [comment]
oh, how about try submitting one or two images (like potato and tomato)! I think the code functions rn in a way that load_trained_model and load_label_binarizer is called only when the user submits something! I can modify the code so that it loads both when the user runs the app instead

**Comment by khmorad** [comment]
Ok I will try that soon I will let you know!

**Comment by tkpp26** [comment]
i modified the code so the vgg16 and label binarizer should hopefully be loaded when u run the app at the beginning!

**Comment by khmorad** [comment]
so i cloned the project again and ran the app.py in back directory but the app.py seem to stop working after giving this error

![image](https://github.com/user-attachments/assets/e0b13f9d-22b3-41b7-91fd-835466c29ce2)

### the .keras file seems to be generated in right location though
![image](https://github.com/user-attachments/assets/6667ec47-03fd-4879-bddb-9b0e62083c52)

if I'm performing something wrong let me know!

**Comment by tkpp26** [comment]
hmm it might be the difference between the version of the tensorflow u have on ur conda env and the one im using. maybe try deactivating conda and creating .venv and pip install -r requirements.txt so that u have all the packages with the same version as mine! i should learn docker haha

**Comment by khmorad** [comment]
thats what i was also thinking! I will try that now!

**Comment by khmorad** [comment]
i get this error when i run the requirements.txt

`ERROR: Ignored the following versions that require a different python version: 1.25.0 Requires-Python >=3.9; 1.25.1 Requires-Python >=3.9; 1.25.2 Requires-Python >=3.9; 1.26.0 Requires-Python <3.13,>=3.9; 1.26.1 Requires-Python <3.13,>=3.9; 1.26.2 Requires-Python >=3.9; 1.26.3 Requires-Python >=3.9; 1.26.4 Requires-Python >=3.9; 2.0.0 Requires-Python >=3.9; 2.0.1 Requires-Python >=3.9; 2.0.2 Requires-Python >=3.9; 2.1.0 Requires-Python >=3.10; 2.1.0rc1 Requires-Python >=3.10; 2.1.1 Requires-Python >=3.10
ERROR: Could not find a version that satisfies the requirement tensorflow==2.17.0 (from versions: 2.2.0, 2.2.1, 2.2.2, 2.2.3, 2.3.0, 2.3.1, 2.3.2, 2.3.3, 2.3.4, 2.4.0, 2.4.1, 2.4.2, 2.4.3, 2.4.4, 2.5.0, 2.5.1, 2.5.2, 2.5.3, 2.6.0rc0, 2.6.0rc1, 2.6.0rc2, 2.6.0, 2.6.1, 2.6.2, 2.6.3, 2.6.4, 2.6.5, 2.7.0rc0, 2.7.0rc1, 2.7.0, 2.7.1, 2.7.2, 2.7.3, 2.7.4, 2.8.0rc0, 2.8.0rc1, 2.8.0, 2.8.1, 2.8.2, 2.8.3, 2.8.4, 2.9.0rc0, 2.9.0rc1, 2.9.0rc2, 2.9.0, 2.9.1, 2.9.2, 2.9.3, 2.10.0rc0, 2.10.0rc1, 2.10.0rc2, 2.10.0rc3, 2.10.0, 2.10.1, 2.11.0rc0, 2.11.0rc1, 2.11.0rc2, 2.11.0, 2.11.1, 2.12.0rc0, 2.12.0rc1, 2.12.0, 2.12.1, 2.13.0rc0, 2.13.0rc1, 2.13.0rc2, 2.13.0, 2.13.1)
ERROR: No matching distribution found for tensorflow==2.17.0`

hmm may I know what version of tensorflow are you using?

**Comment by tkpp26** [comment]
I think its because of the difference between the versions of python we have; i think i have Python 3.10.9. Do u mind checking which version u have by: `python --version`
if thats the case, i can modify the code so that it works with older versions of tensorflow by changing .keras to .h5 to see if it works for u, or u can upgrade the python version! I dont know which would be better, so lmk ur thoughts

**Comment by khmorad** [comment]
im using python 3.12.4 i can try changing to your python version and redo the process again ill let you know!

**Comment by tkpp26** [comment]
oh, im sorry, the issue might have been because the .keras file in my google drive was in "Viewer" mode. I changed it to "Editor" mode just now, let me know if it works!

**Comment by khmorad** [comment]
still get the error which im assuming its my tensor_flow version haha
![image](https://github.com/user-attachments/assets/9672c1e6-990e-43fd-a4bc-2a9543c3d63a)

**Comment by tkpp26** [comment]
hmm..just to make sure u:
1.  changed ur python version
2. deactiavated conda
3. activated .venv and pip installed requriements.txt?

**Comment by tkpp26** [comment]
how about trying changing the .keras to .h5!

**Comment by khmorad** [comment]
changing to .h5 still gives me issues...ill try changing the python version ill let you know

**Comment by tkpp26** [comment]
what's ur tensorflow version?

**Comment by khmorad** [comment]
in my conda environment its 2.3.0

**Comment by tkpp26** [comment]
oh.. mine is tensorflow==2.17.0! So i think its best to deactivate conda for this project, and activate ur .venv and pip install the requirements.txt

**Comment by tkpp26** [comment]
Maybe try pyenv:

`pyenv` allows you to manage multiple versions of Python on the same machine and set up different versions for different projects.

```
pyenv install 3.9.0
pyenv virtualenv 3.9.0 myenv
pyenv activate myenv
pip install -r requirements.txt
```

**Comment by khmorad** [comment]
## yay :D
### so the problem was all on my side once I used python version 3.10.9 and ran the requirements.txt everything worked perfectly thank you for all the help!

<img src="https://github.com/user-attachments/assets/acdf61e2-3207-4d51-a991-05dbd47d3613" alt="Image" width="600">
